lib/iris/mesh/components.py

```diff
@@ -2999,11 +3002,13 @@ def update_from_mesh(self):
         try:
             object.__setattr__(self, "_updating", True)
             if (self._last_modified is None) or (
-                self._last_modified != self.mesh._last_modified
+                self._last_modified < self.mesh._last_modified
             ):
                 points, bounds = self._load_points_and_bounds()
                 super(MeshCoord, self.__class__).points.fset(self, points)
                 super(MeshCoord, self.__class__).bounds.fset(self, bounds)
+                object.__setattr__(self, "_last_modified", self.mesh._last_modified)
+        # Ensure errors aren't bypassed
         except Exception as e:
             raise e
         finally:
```

Comment:
might be worth adding a note to the finally call saying that it ensures _updating isn't left on after a failure

Reply:
I think that's redundant? That's generally what finally does

Reply:
I suppose there's still a case for explaining _why_ it's so important that `_updating` must never be left `True` outside of this context, i.e. that this would break the link between a MeshCoord and it's Mesh and mean that it could no longer be updated.